components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/monetization/DefaultMonetizationImpl.java

```diff
@@ -69,4 +78,28 @@ public boolean disableMonetization(String tenantDomain, API api, Map<String, Str
     public Map<String, String> getTotalRevenue(API api, APIProvider apiProvider) throws MonetizationException {
         return new HashMap<String, String>();
     }
+
+    @Override
+    public boolean publishMonetizationUsageRecords(MonetizationUsagePublishInfo monetizationUsagePublishInfo)
+            throws MonetizationException {
+
+        APIAdmin apiAdmin = new APIAdminImpl();
+        monetizationUsagePublishInfo.setState(APIConstants.MonetizationUsagePublisher.COMPLETED);
+        monetizationUsagePublishInfo.setStatus(APIConstants.MonetizationUsagePublisher.SUCCESSFULL);
+        DateFormat df = new SimpleDateFormat(APIConstants.MonetizationUsagePublisher.TIME_FORMAT);
+        Date dateobj = new Date();
+        //get the time in UTC format
+        df.setTimeZone(TimeZone.getTimeZone(APIConstants.MonetizationUsagePublisher.TIME_ZONE));
+        String currentDate = df.format(dateobj);
+        long currentTimestamp = apiAdmin.getTimestamp(currentDate);
+        monetizationUsagePublishInfo.setLastPublishTime(currentTimestamp);
+        try {
+            apiAdmin.updateMonetizationUsagePublishInfo(monetizationUsagePublishInfo);
+        } catch (APIManagementException e) {
+            String errorMsg = "Failed to update the monetization usage publish info";
```

Comment:
Shall we remove the use of variable and directly use the string value?

Reply:
Fixed in 0cb422c3d49f3c8e3e3d4e02e156665fd004836b